opencti-platform/opencti-graphql/src/migrations/1691746162784-investigation-relations-from-has-reference-to-list-of-entities-and-relations-ids.js

```diff
@@ -0,0 +1,100 @@
+import { elDeleteByQueryForMigration, elUpdateByQueryForMigration } from '../database/engine';
+import {
+  READ_INDEX_INTERNAL_OBJECTS,
+  READ_INDEX_INTERNAL_RELATIONSHIPS,
+  READ_INDEX_STIX_CORE_RELATIONSHIPS,
+  READ_INDEX_STIX_CYBER_OBSERVABLES,
+  READ_INDEX_STIX_DOMAIN_OBJECTS,
+  READ_INDEX_STIX_META_OBJECTS,
+  READ_INDEX_STIX_META_RELATIONSHIPS
+} from '../database/utils';
+import { listAllEntities } from '../database/middleware-loader';
+import { executionContext, SYSTEM_USER } from '../utils/access';
+import { ENTITY_TYPE_WORKSPACE } from '../modules/workspace/workspace-types';
+
+async function fetchPersistedInvestigations() {
+  return await listAllEntities(
+    executionContext('migration'),
+    SYSTEM_USER,
+    [ENTITY_TYPE_WORKSPACE],
+    { type: 'investigation' }
+  );
+}
+
+async function removeInvestigationsReferencesFromInvestigatedEntities(investigationIds) {
+  await elUpdateByQueryForMigration(
+    '[MIGRATION] removing investigations references from investigated entities',
+    [
+      READ_INDEX_STIX_CORE_RELATIONSHIPS,
+      READ_INDEX_STIX_META_RELATIONSHIPS,
+      READ_INDEX_STIX_DOMAIN_OBJECTS,
+      READ_INDEX_STIX_CYBER_OBSERVABLES,
+      READ_INDEX_STIX_META_OBJECTS,
+    ],
+    {
+      script: {
+        source: ''
+          + 'for (def investigationId : params.investigationIds) {'
+          + '  if (ctx._source[params.field].contains(investigationId)) {'
+          + '    ctx._source[params.field].remove(ctx._source[params.field].indexOf(investigationId)) '
```

Comment:
I think you can use `removeAll` directly (I just tested a script and it seems to work) :  

`ctx._source[params.field].removeAll(params.investigationIds)`

Reply:
Thank you for the suggestion. I'm sure it works however, because testing is so tedious, I will just follow the "if it ain't broke, don't fix it" rule. Feel free to modify it though.

Reply:
I'm still thinking about this and a bit worried about the performance of the script on a large database, does someone else have an opinion about it ? but maybe `removeAll` won't really improve the perf, I'm not sure.

Reply:
Should be replace by ctx._source.remove('rel_has-reference') i think